text/0000-volta-run.md

```diff
@@ -0,0 +1,72 @@
+- Feature Name: volta_run
+- Start Date: 2019-09-14
+- RFC PR: (leave this empty)
+- Volta Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add a `volta run` command, allowing users to execute tools with an override on the detected platform.
+
+# Motivation
+[motivation]: #motivation
+
+Volta currently will automatically detect the platform (Node runtime and yarn version) based on the context of where you are in the file system. However, there are occasionally times when you want to run a command using a different version of Node or Yarn than what Volta detects, without changing the pinned or default versions. To support those use-cases, we should provide a command that allows users to execute a command with a platform specified on the command-line.
+
+# Pedagogy
+[pedagogy]: #pedagogy
+
+We will need to document the existence of `volta run` and how the users can specify the versions of Node and Yarn when running a command. This will be provided on the command line by StructOpt, however it should also be explained in detail in the documentation. We should also document how the overall platform is resolved if the user specifies only one of Node or Yarn, so that the behavior is clear to users.
+
+# Details
+[details]: #details
+
+The new command will have the structure:
+
+```
+volta run [--node <version>] [--yarn <version>] <command>
+```
+
+Specifically, the `--node` and `--yarn` settings will need to be specified _before_ the command to be executed, as everything after the command will be treated as arguments to the command itself. This will allow users to specify a full command, with any command-line flags they need, without having to append `--` to separate the `volta run` command from the others.
```

Comment:
I think it's definitely great to not _require_ the `--` separator, but I'd like to suggest that we consider supporting it if used. The `--` as a separator is _pretty_ common, and might feel more natural to folks. I'm a bit torn there because this may just make it more complicated (having to display both versions in docs and stuff) and choosing only one may be best. 

What do you think?

Reply:
Some examples that might be useful here:

```
volta run --node 10 npm run some-script -- --somescript-arg=true
volta run --node 8 --yarn 1.17 yarn some-script --somescript-arg
volta run --node 12 ember build
```

Reply:
I'm not 100% sure, but I think that we will implicitly support the `--`, however we'll still need the `--node`, `--yarn`, etc. to come before the command.

Reply:
@charlespierce - The main thing I was pointing out here is that the `--` in the first example was how `npm` has you pass arguments/options to a `npm run-script`.